### PR TITLE
Use bintray artifactory

### LINKED ISSUE #22: Update artifactory to bintray
- Update artifactory to use bintray
- Also we are using maven 3 which does not require a plugin repo. I think that section can be removed.

### PR DESCRIPTION
**Pull request by yqian30**
Make changes to use bintray artifactory.
This is to fix #22

### CODE REVIEW

**Comment by mrmccarthy** [comment]
I see this when  building on EMC Open after clearing out my .m2 repository

```
[INFO] ------------------------------------------------------------------------
[INFO] Building DSB Java Web API 0.38
[INFO] ------------------------------------------------------------------------
Downloading: https://repo.maven.apache.org/maven2/io/swagger/swagger-codegen-maven-plugin/2.2.4-OCOPEA/swagger-codegen-maven-plugin-2.2.4-OCOPEA.pom
             
[WARNING] The POM for io.swagger:swagger-codegen-maven-plugin:jar:2.2.4-OCOPEA is missing, no dependency information available
Downloading: https://repo.maven.apache.org/maven2/io/swagger/swagger-codegen-maven-plugin/2.2.4-OCOPEA/swagger-codegen-maven-plugin-2.2.4-OCOPEA.jar
             
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Parent project for Ocopea Dev Tools ................ SUCCESS [ 25.176 s]
[INFO] Ocopea checkstyle definitions ...................... SUCCESS [01:02 min]
[INFO] Orcs Parent project ................................ SUCCESS [  0.018 s]
[INFO] Utilities Parent ................................... SUCCESS [  0.021 s]
[INFO] Common Utility ..................................... SUCCESS [ 35.771 s]
[INFO] Native Query Service ............................... SUCCESS [  8.870 s]
[INFO] DB JUnit Support ................................... SUCCESS [  0.166 s]
[INFO] Simple workflow library ............................ SUCCESS [  1.121 s]
[INFO] JSON utilities ..................................... SUCCESS [  9.086 s]
[INFO] Postgres related Utilities ......................... SUCCESS [ 12.283 s]
[INFO] Rest Utilities ..................................... SUCCESS [ 16.443 s]
[INFO] Persistent  Messaging Parent ....................... SUCCESS [  0.016 s]
[INFO] Shpan Messaging API ................................ SUCCESS [  0.121 s]
[INFO] Persistent Messaging Implementation ................ SUCCESS [ 35.176 s]
[INFO] Postgres Message Persister ......................... SUCCESS [ 14.100 s]
[INFO] Persistent Scheduler Parent ........................ SUCCESS [  0.016 s]
[INFO] Persistent Scheduler API ........................... SUCCESS [  0.062 s]
[INFO] Persistent Scheduler Implementation ................ SUCCESS [ 59.378 s]
[INFO] Java Micro Service Library API ..................... SUCCESS [  4.607 s]
[INFO] Java Micro Service Library ......................... SUCCESS [  0.002 s]
[INFO] Micro Service Scheduler API ........................ SUCCESS [  0.072 s]
[INFO] Micro Service Configuration API .................... SUCCESS [  0.083 s]
[INFO] DPA Service Metrics Registry ....................... SUCCESS [  5.651 s]
[INFO] DPA Micro Service Timer ............................ SUCCESS [ 11.423 s]
[INFO] DPA Microservice Serialization Utils ............... SUCCESS [  0.169 s]
[INFO] Micro Service Runtime .............................. SUCCESS [ 13.752 s]
[INFO] DPA Micro Service Object Store ..................... SUCCESS [  0.008 s]
[INFO] object-store-temp-filesystem-impl .................. SUCCESS [  3.684 s]
[INFO] DPA Micro Service Schema Bootstrap ................. SUCCESS [  0.008 s]
[INFO] DPA Micro Service Bootstrap Schema ................. SUCCESS [  1.056 s]
[INFO] DPA Micro Service Testing Framework ................ SUCCESS [ 13.189 s]
[INFO] DPA Micro Service Web API .......................... SUCCESS [  0.344 s]
[INFO] DPA Micro Service REST Resources ................... SUCCESS [  7.878 s]
[INFO] DPA Micro Service JMS Messaging .................... SUCCESS [  2.196 s]
[INFO] DPA Micro Service RabbitMQ Messaging ............... SUCCESS [ 14.060 s]
[INFO] DPA Micro Service Standalone Web Server ............ SUCCESS [  8.003 s]
[INFO] DPA Micro Service Undertow Web Server .............. SUCCESS [ 24.135 s]
[INFO] Standalone Postgres Datasource Implementation ...... SUCCESS [  3.320 s]
[INFO] DPA Micro Service Object Store PGSQL implementation  SUCCESS [  1.330 s]
[INFO] DPA Micro Service Object Store PGSQL Standalone .... SUCCESS [  0.120 s]
[INFO] Standalone Postgres Blob Store Implementation ...... SUCCESS [  0.166 s]
[INFO] DPA Micro Service Messaging Stats Resource Manager . SUCCESS [  0.156 s]
[INFO] Dev Stack Providers Parent ......................... SUCCESS [  0.026 s]
[INFO] Dev Blob Store Provider ............................ SUCCESS [  0.179 s]
[INFO] Dev Data Source Provider ........................... SUCCESS [  0.125 s]
[INFO] Dev Messaging Provider ............................. SUCCESS [  0.066 s]
[INFO] Dev Web Server Provider ............................ SUCCESS [  0.029 s]
[INFO] Dev Scheduler Provider ............................. SUCCESS [  0.051 s]
[INFO] Logback Logging Provider ........................... SUCCESS [  0.123 s]
[INFO] DEV Deployment Stack ............................... SUCCESS [  1.428 s]
[INFO] DPA Micro Service Samples and integration tests .... SUCCESS [  0.002 s]
[INFO] Calculator Microservice Sample Parent Project ...... SUCCESS [  0.025 s]
[INFO] Calculator Microservice Sample Public Model ........ SUCCESS [  0.076 s]
[INFO] Calculator Sample Micro-Service Implementation ..... SUCCESS [  2.654 s]
[INFO] Calculator Sample Development Deployer ............. SUCCESS [ 34.714 s]
[INFO] Banking Samples Parent Project ..................... SUCCESS [  0.013 s]
[INFO] Sample BankDB schema bootstrap ..................... SUCCESS [  0.080 s]
[INFO] Sample Banking Common Model ........................ SUCCESS [  0.053 s]
[INFO] Sample Banking Common Services ..................... SUCCESS [  0.105 s]
[INFO] Deposit to Account Sample Parent Project ........... SUCCESS [  0.009 s]
[INFO] Deposit Microservice Sample Public Model ........... SUCCESS [  0.047 s]
[INFO] Deposit Sample Micro-Service Implementation ........ SUCCESS [  1.995 s]
[INFO] Sample Bank UI Server .............................. SUCCESS [  0.016 s]
[INFO] Deposit Sample Micro-Service Web API ............... SUCCESS [  0.062 s]
[INFO] Deposit Sample Micro-Service Implementation ........ SUCCESS [  2.179 s]
[INFO] Sample Banking Integration Tests ................... SUCCESS [ 15.048 s]
[INFO] Inspector Parent ................................... SUCCESS [  0.007 s]
[INFO] Inspector public model ............................. SUCCESS [  0.138 s]
[INFO] Inspector Web API .................................. SUCCESS [  0.073 s]
[INFO] Inspector Micro Service ............................ SUCCESS [  0.316 s]
[INFO] Microservice Sample Dev Runner ..................... SUCCESS [  0.174 s]
[INFO] Microservice Sample Platform Test .................. SUCCESS [ 57.526 s]
[INFO] sample-schedule .................................... SUCCESS [  0.006 s]
[INFO] object-store-pgsql-schema-bootstrap ................ SUCCESS [  2.805 s]
[INFO] DPA Micro Service Object Store Implementation ...... SUCCESS [ 24.909 s]
[INFO] Configuration Microservice Parent .................. SUCCESS [  0.004 s]
[INFO] Configuration Web API .............................. SUCCESS [  0.104 s]
[INFO] Configuration Bootstrap ............................ SUCCESS [  0.090 s]
[INFO] Configuration Service .............................. SUCCESS [  4.092 s]
[INFO] Configuraiton Service Dev Runner ................... SUCCESS [  0.373 s]
[INFO] Configuration Service Client ....................... SUCCESS [  4.545 s]
[INFO] Remote DEV Deployment Stack Parent ................. SUCCESS [  0.002 s]
[INFO] Remote DEV Resource Provider ....................... SUCCESS [  0.330 s]
[INFO] Remote DEV Deployment Stack Server Parent .......... SUCCESS [  0.002 s]
[INFO] Remote DEV Deployment Stack Server Web API ......... SUCCESS [  0.067 s]
[INFO] Remote DEV Deployment Stack Server ................. SUCCESS [  0.170 s]
[INFO] Remote DEV Deployment Stack Server Deployer ........ SUCCESS [  0.185 s]
[INFO] Undertow Web Server Provider ....................... SUCCESS [  0.148 s]
[INFO] Registry client external resource manager .......... SUCCESS [  0.005 s]
[INFO] Registry client external RM api .................... SUCCESS [  0.071 s]
[INFO] Registry client external RM implementation ......... SUCCESS [  0.153 s]
[INFO] Micro Service Tests ................................ SUCCESS [  7.630 s]
[INFO] Persistent Messaging Provider ...................... SUCCESS [  0.166 s]
[INFO] Persistent Scheduler Provider ...................... SUCCESS [  0.121 s]
[INFO] Parent project for DSB ............................. SUCCESS [  0.011 s]
[INFO] DSB Web API ........................................ SUCCESS [  0.050 s]
[INFO] DSB Java Web API ................................... FAILURE [  3.256 s]
[INFO] Parent project for copy repository ................. SKIPPED
[INFO] Copy Repository Web API ............................ SKIPPED
[INFO] Copy Repository Java Web API ....................... SKIPPED
[INFO] Parent project for PSB ............................. SKIPPED
[INFO] PSB Java Web API ................................... SKIPPED
[INFO] Site Parent ........................................ SKIPPED
[INFO] Site Web API ....................................... SKIPPED
[INFO] Site Repository parent ............................. SKIPPED
[INFO] Site Repository Schema ............................. SKIPPED
[INFO] Site Repository API ................................ SKIPPED
[INFO] Artifact registry .................................. SKIPPED
[INFO] Parent project for protection services ............. SKIPPED
[INFO] Protection Policy Java Web API ..................... SKIPPED
[INFO] Site Domain ........................................ SKIPPED
[INFO] Site Repository Implementations .................... SKIPPED
[INFO] Common tools for CRB ............................... SKIPPED
[INFO] Protection Policy repository parent project ........ SKIPPED
[INFO] Protection Repository API .......................... SKIPPED
[INFO] Protection Domain Aggregates ....................... SKIPPED
[INFO] Protection Repository Schema ....................... SKIPPED
[INFO] Protection Repository Implementation ............... SKIPPED
[INFO] Protection Policy Service .......................... SKIPPED
[INFO] Site MicroService .................................. SKIPPED
[INFO] Site Remote Dev Runner ............................. SKIPPED
[INFO] Site Code Coverage ................................. SKIPPED
[INFO] nui parent ......................................... SKIPPED
[INFO] ui ................................................. SKIPPED
[INFO] nui ................................................ SKIPPED
[INFO] nui login .......................................... SKIPPED
[INFO] nui hackathon app .................................. SKIPPED
[INFO] nui hackathon committee app ........................ SKIPPED
[INFO] Hub Parent project ................................. SKIPPED
[INFO] Hub Util ........................................... SKIPPED
[INFO] Hub Service Parent ................................. SKIPPED
[INFO] Hub Web API ........................................ SKIPPED
[INFO] Hub Repository Parent project ...................... SKIPPED
[INFO] Hub Repository API ................................. SKIPPED
[INFO] Hub Repository Schema .............................. SKIPPED
[INFO] Hub Repository Implementation ...................... SKIPPED
[INFO] Hub Service ........................................ SKIPPED
[INFO] Hub Web App Parent ................................. SKIPPED
[INFO] Hub WebApp Web API ................................. SKIPPED
[INFO] Hub WebApp Java Web API ............................ SKIPPED
[INFO] Hub Auth Util ...................................... SKIPPED
[INFO] Hub WebApp Service ................................. SKIPPED
[INFO] Shpan Copy Repo Reference Impl Parent .............. SKIPPED
[INFO] Shpan Copy Repo Service ............................ SKIPPED
[INFO] Shpan Copy Repo Tester ............................. SKIPPED
[INFO] Shpan Copy Repo Reference implementation Remote Dev Runner SKIPPED
[INFO] orcs-deployer-parent ............................... SKIPPED
[INFO] Ocopea orcs docker runner .......................... SKIPPED
[INFO] orcs-docker-image .................................. SKIPPED
[INFO] Ocopea Test Scenarios .............................. SKIPPED
[INFO] Orcs Single Jar parent ............................. SKIPPED
[INFO] ShpanPaaS PSB Parent ............................... SKIPPED
[INFO] Hackathon Services ................................. SKIPPED
[INFO] ShpanPaaS Resource Provider ........................ SKIPPED
[INFO] ShpanPaaS PSB Service .............................. SKIPPED
[INFO] H2 DSB Parent ...................................... SKIPPED
[INFO] ShpanPaaS H2 DSB ................................... SKIPPED
[INFO] ShpanBlob DSB ...................................... SKIPPED
[INFO] ShpanPaaS PSB Remote Dev Runner .................... SKIPPED
[INFO] Ocopea Orcs Single Jar Runner ...................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 09:28 min
[INFO] Finished at: 2017-10-26T10:19:05-04:00
[INFO] Final Memory: 102M/244M
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin io.swagger:swagger-codegen-maven-plugin:2.2.4-OCOPEA or one of its dependencies could not be resolved: Could not find artifact io.swagger:swagger-codegen-maven-plugin:jar:2.2.4-OCOPEA in central (https://repo.maven.apache.org/maven2) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException

```

**Comment by yqian30** [comment]
The pluginRepository is still needed. Put it back . @mrmccarthy @vjtirumalai

File: pom.xml [hunk 1]
```diff
@@ -332,41 +332,31 @@
...
     </profiles>
+    <pluginRepositories>
+         <!-- our public bintray repository-->
+        <pluginRepository>
+             <id>ocopea-dependencies</id>
+             <name>bintray-plugins</name>
+              <snapshots>
+                  <enabled>false</enabled>
+              </snapshots>
+             <url>https://raw.githubusercontent.com/ocopea/artifactory/repository/</url>
```
Comment (vjtirumalai) [inline pom.xml]: Why is it github and not bintray?

Comment (vjtirumalai) [reply pom.xml]: Fixed it myself. Builds fine now.

**Pull request merged**